### PR TITLE
Repeat panels when row is expanding

### LINKED ISSUE #10647: Not update expand row content
1. Repeat row (variable1) with repeat graph (variable2).
2. Collapse row.
3. Change variable2 value.
4. Expand row.

Repeat graphs are not changed.

Grafana version: v5.0.0-pre1 (commit: c34d458)
Datasource: TestData
Server OS: Ubuntu 16.04.3 LTS (4.4.0-72-generic)
Client OS: Ubuntu 17.10 (4.13.0-21-generic)
Browser: Google Chrome 63.0.3239.132

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This PR fixes #10647

### CODE REVIEW

File: public/app/features/dashboard/specs/repeat.jest.ts [hunk 1]
```diff
@@ -4,6 +4,57 @@ import { expect } from 'test/lib/common';
...
+            ],
+          },
+        ],
+      },
+    };
+    dashboard = new DashboardModel(dashboardJSON);
+    dashboard.processRepeats();
+  });
+
+  it('should repeat panels when row is collapsed', function() {
```
Comment (torkelo) [inline public/app/features/dashboard/specs/repeat.jest.ts]: should this not be when row is expanded?

**Comment by torkelo** [comment]
@ilgizar I think this is not trying to solve that problem, it is only addressing https://github.com/grafana/grafana/issues/10647 which deals with what happens when you expand a row.

File: public/app/features/dashboard/specs/repeat.jest.ts (hunk 1 above)
Comment (ilgizar) [reply public/app/features/dashboard/specs/repeat.jest.ts]: My mistake, should be the word "expanded".

Comment (alexanderzobnin) [reply public/app/features/dashboard/specs/repeat.jest.ts]: Yes, should rename this.

**Comment by ilgizar** [comment]
Why you merge this PR? I warned yesterday that such a method would not work. More precisely, it works only for the first row.
Yes, my PRs are cumbersome and not a simple and clear as @alexanderzobnin's "sharp knife". But I still see the shortcomings, and this was not the defense of my option. How many people have so many opinions.
I apologize if I was intrusive and English of google translator.
After applying this PR:
1. Repeat row by var1 [reg1,reg2]
2. Repeat panel inside row by var2 [ser1,ser2] with title "$var1 $var2"
3. Сollapse both rows
4. Try to expand second row. **First try - no changes. Second - empty row.**
5. Expand first row - all row expanded. Tolerant.

PS: further there is an incomprehensible mistake for me. it is possible to repeat only with me.
6. Collapse and expand only second row. Second panel in first row have title "reg1 + reg2 ser1 + ser2".

**Comment by alexanderzobnin** [comment]
@ilgizar yes, confirmed, my fix doesn't work for the second row. I prepared fix and think about how to save row state while repeats are processed.

**Comment by alexanderzobnin** [comment]
That's really tricky stuff due to repeats processing design. I should improve it.

**Pull request merged**